BUG: invalid memory address or nil pointer dereference

# Problem Description

_INSTRUCTIONS: Describe the problem in as much detail as possible, including the actions which caused the problem, and any other information you believe would be helpful in reproducing the problem._

``
A crash occurred
``

# System Information

_INSTRUCTIONS: Provide the system and application information below._

CLUSTERAPI VERSION: ` v1.9.6 `
SVELTOS VERSION: ` v0.52.2`
KUBERNETES VERSION: ` v1.31.7-eks-bcf3d70 `

# Logs

_INSTRUCTIONS: Provide any additional information you think would be helpful below.  Large files, logs, etc. can be attached to this issue so long as they meet the GitHub attachment guidelines described here: https://help.github.com/en/github/managing-your-work-on-github/file-attachments-on-issues-and-pull-requests_

```
I0428 00:47:20.766082       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="7ea73017-18dc-4e15-875c-74a08ef5e29a"
E0428 00:47:20.767134       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="7ea73017-18dc-4e15-875c-74a08ef5e29a"
I0428 00:47:20.774166       1 handler.go:53] "v1beta2 Bucket is deprecated, upgrade to v1" logger="API Server Warning"
I0428 00:47:20.775660       1 handler.go:53] "v1beta2 Bucket is deprecated, upgrade to v1" logger="API Server Warning"
I0428 00:47:20.789000       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="1" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:20.810073       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="c218d025-88db-4a8e-af79-b5257416b7f4"
I0428 00:47:20.810576       1 clusterprofile_controller.go:94] "Reconciling" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="9e748b74-b946-4cb1-8ad7-ac92a4cbe7e2"
I0428 00:47:20.810603       1 clusterprofile_controller.go:155] "Reconciling ClusterProfile" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="9e748b74-b946-4cb1-8ad7-ac92a4cbe7e2"
I0428 00:47:20.812769       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="e88a2010-cc64-4f50-aa1d-cf72f3db2fbb"
I0428 00:47:20.813041       1 clusterprofile_controller.go:94] "Reconciling" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="629fe98b-e803-4073-9a28-0c1af80dc9f7"
I0428 00:47:20.813069       1 clusterprofile_controller.go:155] "Reconciling ClusterProfile" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="629fe98b-e803-4073-9a28-0c1af80dc9f7"
I0428 00:47:20.813080       1 cluster_utils.go:535] "empty selector matches no cluster" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="629fe98b-e803-4073-9a28-0c1af80dc9f7"
I0428 00:47:20.848906       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="9e748b74-b946-4cb1-8ad7-ac92a4cbe7e2"
I0428 00:47:20.855483       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="629fe98b-e803-4073-9a28-0c1af80dc9f7"
I0428 00:47:20.866387       1 clusterprofile_controller.go:94] "Reconciling" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="a4f59beb-9c10-4fc4-aaa3-6766057c2618"
I0428 00:47:20.866446       1 clusterprofile_controller.go:155] "Reconciling ClusterProfile" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="a4f59beb-9c10-4fc4-aaa3-6766057c2618"
I0428 00:47:20.866479       1 cluster_utils.go:535] "empty selector matches no cluster" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="a4f59beb-9c10-4fc4-aaa3-6766057c2618"
I0428 00:47:20.866396       1 clusterprofile_controller.go:94] "Reconciling" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="737454b9-d3df-465a-9e0c-d13c29212e7b"
I0428 00:47:20.866631       1 clusterprofile_controller.go:155] "Reconciling ClusterProfile" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="737454b9-d3df-465a-9e0c-d13c29212e7b"
I0428 00:47:20.910840       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="a4f59beb-9c10-4fc4-aaa3-6766057c2618"
I0428 00:47:20.911125       1 clusterprofile_controller.go:94] "Reconciling" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="92448433-07c2-493e-91ef-ba3fea8cb7ab"
I0428 00:47:20.911169       1 clusterprofile_controller.go:155] "Reconciling ClusterProfile" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="92448433-07c2-493e-91ef-ba3fea8cb7ab"
I0428 00:47:20.911182       1 cluster_utils.go:535] "empty selector matches no cluster" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="92448433-07c2-493e-91ef-ba3fea8cb7ab"
I0428 00:47:20.934968       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="k0smotron-karpenter-resources" namespace="" name="k0smotron-karpenter-resources" reconcileID="737454b9-d3df-465a-9e0c-d13c29212e7b"
I0428 00:47:20.949430       1 clusterprofile_controller.go:194] "Reconcile success" controller="clusterprofile" controllerGroup="config.projectsveltos.io" controllerKind="ClusterProfile" ClusterProfile="karpenter-kahiro" namespace="" name="karpenter-kahiro" reconcileID="92448433-07c2-493e-91ef-ba3fea8cb7ab"
I0428 00:47:30.687683       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="92db942b-dabf-4dec-af11-489b5b6a7ea7"
I0428 00:47:30.687827       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="92db942b-dabf-4dec-af11-489b5b6a7ea7"
E0428 00:47:30.688524       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: request is queued" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="92db942b-dabf-4dec-af11-489b5b6a7ea7"
I0428 00:47:30.701611       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="4e7cf2f0-db35-4306-bcf2-6deda167d2a3"
I0428 00:47:30.702957       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="4e7cf2f0-db35-4306-bcf2-6deda167d2a3"
E0428 00:47:30.705737       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="4e7cf2f0-db35-4306-bcf2-6deda167d2a3"
I0428 00:47:30.727585       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="8ffdcba5-e7ae-4ee5-b384-f66dd92e9bc9"
I0428 00:47:30.727656       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="8ffdcba5-e7ae-4ee5-b384-f66dd92e9bc9"
E0428 00:47:30.728425       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="8ffdcba5-e7ae-4ee5-b384-f66dd92e9bc9"
I0428 00:47:30.748859       1 worker.go:172] "worker: 11 processing request. cleanup: false" logger="deployer" worker="11" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false"
I0428 00:47:30.749451       1 cluster_cache.go:132] "remote restConfig cache hit" logger="deployer" worker="11" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/"
I0428 00:47:30.749682       1 handlers_utils.go:322] "referencedObject ConfigMap mgmt/k0smotron-karpenter-resources-template-remote is a template" logger="deployer" worker="11" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:30.749696       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="11" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:30.781659       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="11" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:40.702170       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="a63b0fd1-c199-41d0-86c3-d3cc9c217db4"
I0428 00:47:40.702308       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="a63b0fd1-c199-41d0-86c3-d3cc9c217db4"
E0428 00:47:40.703316       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: request is queued" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="a63b0fd1-c199-41d0-86c3-d3cc9c217db4"
I0428 00:47:40.716856       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="ae886314-71c4-45fa-912e-a98de2d38dc5"
I0428 00:47:40.716950       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="ae886314-71c4-45fa-912e-a98de2d38dc5"
E0428 00:47:40.717758       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="ae886314-71c4-45fa-912e-a98de2d38dc5"
I0428 00:47:40.729288       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="7b847c87-9b76-4e20-b092-562848bd42fb"
I0428 00:47:40.729355       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="7b847c87-9b76-4e20-b092-562848bd42fb"
E0428 00:47:40.730010       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="7b847c87-9b76-4e20-b092-562848bd42fb"
I0428 00:47:40.753250       1 worker.go:172] "worker: 15 processing request. cleanup: false" logger="deployer" worker="15" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false"
I0428 00:47:40.753994       1 cluster_cache.go:132] "remote restConfig cache hit" logger="deployer" worker="15" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/"
I0428 00:47:40.754216       1 handlers_utils.go:322] "referencedObject ConfigMap mgmt/k0smotron-karpenter-resources-template-remote is a template" logger="deployer" worker="15" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:40.754232       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="15" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:40.783433       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="15" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:50.717425       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="bc62c97e-30a7-4ed4-bd1e-3a0009a7a5bd"
I0428 00:47:50.717612       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="bc62c97e-30a7-4ed4-bd1e-3a0009a7a5bd"
E0428 00:47:50.718556       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: request is queued" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="bc62c97e-30a7-4ed4-bd1e-3a0009a7a5bd"
I0428 00:47:50.733408       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="d99f6019-9f4c-4e0d-8742-5a1f0cdc5816"
I0428 00:47:50.733543       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="d99f6019-9f4c-4e0d-8742-5a1f0cdc5816"
E0428 00:47:50.735644       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="d99f6019-9f4c-4e0d-8742-5a1f0cdc5816"
I0428 00:47:50.747727       1 clustersummary_controller.go:131] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="05c0b3f4-9ba1-477b-b510-b2c7e7fbda1c"
I0428 00:47:50.747827       1 clustersummary_controller.go:326] "Reconciling ClusterSummary" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="05c0b3f4-9ba1-477b-b510-b2c7e7fbda1c"
E0428 00:47:50.748671       1 clustersummary_controller.go:390] "failed to deploy" err="deploying resources failed: feature is still being provisioned" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="k0smotron-kahiro/k0smotron-karpenter-resources-capi-kahiro" namespace="k0smotron-kahiro" name="k0smotron-karpenter-resources-capi-kahiro" reconcileID="05c0b3f4-9ba1-477b-b510-b2c7e7fbda1c"
I0428 00:47:50.758760       1 worker.go:172] "worker: 5 processing request. cleanup: false" logger="deployer" worker="5" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false"
I0428 00:47:50.759494       1 cluster_cache.go:132] "remote restConfig cache hit" logger="deployer" worker="5" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/"
I0428 00:47:50.759738       1 handlers_utils.go:322] "referencedObject ConfigMap mgmt/k0smotron-karpenter-resources-template-remote is a template" logger="deployer" worker="5" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:50.759755       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="5" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
I0428 00:47:50.789869       1 template_instantiation.go:111] "Fetch cluster Capi: k0smotron-kahiro/kahiro" logger="deployer" worker="5" key="k0smotron-kahiro:::kahiro:::Capi:::k0smotron-karpenter-resources-capi-kahiro:::Resources:::false" cluster="k0smotron-kahiro/kahiro" clusterSummary="k0smotron-karpenter-resources-capi-kahiro" admin="/" configMapNamespace="mgmt" configMapName="k0smotron-karpenter-resources-template-remote"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc7704a]

goroutine 95 [running]:
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).SetManagedFields(0x2b13a60?, {0x0?, 0x2b385a5?, 0xb?})
    /go/pkg/mod/k8s.io/apimachinery@v0.32.3/pkg/apis/meta/v1/unstructured/unstructured.go:480 +0x2a
github.com/projectsveltos/addon-controller/controllers.omitManagedFields(0x0)
    /workspace/controllers/handlers_utils.go:789 +0x48
github.com/projectsveltos/addon-controller/controllers.evaluateResourceDiff(0xc000538868, 0x0)
    /workspace/controllers/handlers_utils.go:738 +0x2b1
github.com/projectsveltos/addon-controller/controllers.generateResourceReport({0xc003cab6d0, 0x47}, 0xc005abadb0, 0x0, 0xc0042641a0)
    /workspace/controllers/handlers_utils.go:716 +0x150
github.com/projectsveltos/addon-controller/controllers.deployUnstructured({0x2f5d3e8, 0xc0006aaaa0}, 0x1, 0xc001a43208, {0x2f796a0, 0xc000299d40}, {0xc000ead6f0, 0x2, 0x2}, 0xc001a7b650, ...)
    /workspace/controllers/handlers_utils.go:547 +0xcc5
github.com/projectsveltos/addon-controller/controllers.deployContent({0x2f5d3e8, 0xc0006aaaa0}, 0x1, 0xc001a43208, {0x2f796a0, 0xc000299d40}, {0x2f9b0b0, 0xc001a41b00}, 0xc0019f07e0, 0xc001a32308, ...)
    /workspace/controllers/handlers_utils.go:381 +0x379
github.com/projectsveltos/addon-controller/controllers.deployContentOfConfigMap(...)
    /workspace/controllers/handlers_utils.go:120
github.com/projectsveltos/addon-controller/controllers.deployObjects({0x2f5d3e8, 0xc0006aaaa0}, 0x1, {0x2f796a0, 0xc000299d40}, 0xc001a43208, {0xc004992ee0, 0x1, 0xc00198faa0?}, 0xc001a32308, ...)
    /workspace/controllers/handlers_utils.go:1247 +0x3d1
github.com/projectsveltos/addon-controller/controllers.deployReferencedObjects({0x2f5d3e8, 0xc0006aaaa0}, {0x2f796a0, 0xc000299d40}, 0xc006c5f448, 0xc001a32308, {0xc004992ee0, 0x1, 0x2}, {0xc004992f00, ...}, ...)
    /workspace/controllers/handlers_utils.go:1215 +0x335
github.com/projectsveltos/addon-controller/controllers.deployPolicyRefs({0x2f5d3e8, 0xc0006aaaa0}, {0x2f796a0, 0xc000299d40}, 0xc006c5f448, 0xc001a32308, {{0x2b34cf9, 0x9}, 0x2c78430, 0x2c783b8, ...}, ...)
    /workspace/controllers/handlers_resources.go:510 +0x166
github.com/projectsveltos/addon-controller/controllers.deployResources({0x2f5d3e8, 0xc0006aaaa0}, {0x2f796a0, 0xc000299d40}, {0xc004189980, 0x10}, {0xc004189993, 0x6}, {0xc0041899a3, 0x29}, ...)
    /workspace/controllers/handlers_resources.go:70 +0x2f2
github.com/projectsveltos/addon-controller/controllers.genericDeploy({0x2f5d3e8, 0xc0006aaaa0}, {0x2f796a0, 0xc000299d40}, {0xc004189980, 0x10}, {0xc004189993, 0x6}, {0xc0041899a3, 0x29}, ...)
    /workspace/controllers/clustersummary_deployer.go:195 +0x1a4
github.com/projectsveltos/libsveltos/lib/deployer.processRequests({0x2f5d3e8, 0xc0006aaaa0}, 0xc00025f4d0, 0x5, {{0x2f669e0?, 0xc0006a9800?}, 0x0?})
    /go/pkg/mod/github.com/projectsveltos/libsveltos@v0.52.2/lib/deployer/worker.go:175 +0x4d3
created by github.com/projectsveltos/libsveltos/lib/deployer.(*deployer).startWorkloadWorkers in goroutine 1
    /go/pkg/mod/github.com/projectsveltos/libsveltos@v0.52.2/lib/deployer/worker.go:87 +0x17b
stream closed EOF for projectsveltos/addon-controller-5b5bdf49bd-zb6kn (controller)
```

![Image](https://github.com/user-attachments/assets/324c5e22-c26b-41f7-8bdb-44520731e961)

--- a/controllers/clustersummary_watchers.go
+++ b/controllers/clustersummary_watchers.go
@@ -417,6 +417,8 @@ func (m *manager) react(obj client.Object, logger logr.Logger) {
 		Name:       obj.GetName(),
 	}
 
+	logger = logger.WithValues("resource", fmt.Sprintf("%s/%s", ref.Namespace, ref.Name))
+
 	// finds all ClusterSummary objects that want to be informed about updates to this specific resource.
 	// It takes into account registrations made through KustomizationRefs
 	if v, ok := m.mgmtResourcesWatchedKustomizeRef[ref]; ok {

--- a/controllers/handlers_utils.go
+++ b/controllers/handlers_utils.go
@@ -543,8 +543,10 @@ func deployUnstructured(ctx context.Context, deployingToMgmtCluster bool, destCo
 			policy.GetKind(), policy.GetNamespace(), policy.GetName(), deployingToMgmtCluster))
 
 		updatedPolicy, err := updateResource(ctx, dr, clusterSummary, policy, subresources, logger)
-		resource.LastAppliedTime = &metav1.Time{Time: time.Now()}
-		reports = append(reports, *generateResourceReport(policyHash, resourceInfo, updatedPolicy, resource))
+		if updatedPolicy != nil {
+			resource.LastAppliedTime = &metav1.Time{Time: time.Now()}
+			reports = append(reports, *generateResourceReport(policyHash, resourceInfo, updatedPolicy, resource))
+		}
 		if err != nil {
 			if clusterSummary.Spec.ClusterProfileSpec.ContinueOnError {
 				errorMsg += fmt.Sprintf("%v", err)

